src/sympc/protocol/falcon/falcon.py

```diff
@@ -404,3 +405,52 @@ def multiplication_protocol(
             x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
+        """Returns either x or y based on bit b.
+
+        Args:
+            x (MPCTensor): input tensor
+            y (MPCTensor): input tensor
+            b (MPCTensor): input tensor which is shares of a bit used as selector bit.
+
+        Returns:
+            z (MPCTensor):Returns x (if b==0) or y (if b==1).
+
+        Raises:
+            ValueError: If the selector bit tensor is not of ring size "2".
+        """
+        ring_size = int(b.share_ptrs[0].get_ring_size().get_copy())
+        shape = b.shape
+        if ring_size != 2:
+            raise ValueError(
+                f"Invalid {ring_size} for bit_injection,must be of ring size 2"
```

Comment:
You mean ring size of selection bit right?

Reply:
That would be appropriate , I would change it :+1:

Reply:
Done